receiver/receiverprofiles/factory.go

```diff
@@ -0,0 +1,26 @@
+package receiverprofiles
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/internal"
+)
+
+type Profiles = internal.Profiles
+
+// Factory is factory interface for receivers that can create profiles receivers.
+// It is an optional interface that receivers can implement if they support profiles.
+type Factory interface {
+	CreateProfilesReceiver(ctx context.Context, set receiver.CreateSettings, cfg component.Config, nextConsumer consumer.Profiles) (Profiles, error)
+	internal.Internal
+}
+
+type CreateProfilesFunc = internal.CreateProfilesFunc
+
+// WithProfiles adds support for profiling to a receiver factory.
+func WithProfiles(createProfilesReceiver CreateProfilesFunc, sl component.StabilityLevel) receiver.FactoryOption {
+	return internal.WithProfiles(createProfilesReceiver, sl)
```

Comment:
Won't the compiler complain that `receiverprofiles.CreateProfilesFunc` is not the same type as `internal.CreateProfilesFunc`?

Reply:
On 0a86ed60b28100ded4b861701d0e6443f65ce46e I added a usage of `WithProfiles` in the otlpreceiver to show that it works https://github.com/open-telemetry/opentelemetry-collector/pull/10253/commits/0a86ed60b28100ded4b861701d0e6443f65ce46e?file-filters%5B%5D=.go&show-viewed-files=true I can build the Collector. I think that should show it works, right?